packages/base/adapters/solana/web3js/client.ts

```diff
@@ -175,59 +180,67 @@ export class Web3Modal extends Web3ModalScaffold {
 
       formatUnits: () => ''
     }
+  }
+
+  public construct(appKit: AppKit, options: OptionsControllerState) {
+    const { projectId } = options
+    const clientOptions = this.instanceOptions
+
+    if (!clientOptions) {
+      throw new Error('Solana:construct - clientOptions is undefined')
+    }
 
-    super({
-      chain: CommonConstantsUtil.CHAIN.SOLANA,
-      networkControllerClient,
-      connectionControllerClient,
-      supportedWallets: wallets,
-
-      defaultChain: SolHelpersUtil.getChainFromCaip(
-        chains,
-        typeof window === 'object' ? localStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) : ''
-      ) as CaipNetwork,
-      tokens: HelpersUtil.getCaipTokens(tokens),
-      _sdkVersion: _sdkVersion ?? `html-solana-${ConstantsUtil.VERSION}`,
-      ...w3mOptions
-    } as ScaffoldOptions)
+    this.appKit = appKit
+    this.options = options
+
+    const { chains } = clientOptions
+
+    if (!projectId) {
+      throw new Error('Solana:construct - projectId is undefined')
+    }
 
     this.initializeProviders({
       relayUrl: 'wss://relay.walletconnect.com',
-      metadata,
-      projectId: w3mOptions.projectId
+      metadata: clientOptions.metadata,
+      projectId: options.projectId
     })
 
-    this.chains = chains
-    this.connectionSettings = connectionSettings
-    this.syncRequestedNetworks(chains, chainImages)
+    this.syncRequestedNetworks(chains, this.options?.chainImages)
 
     const chain = SolHelpersUtil.getChainFromCaip(
       chains,
       typeof window === 'object' ? localStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) : ''
     )
+
+    this.defaultChain = SolHelpersUtil.getChainFromCaip(
+      chains,
+      typeof window === 'object' ? localStorage.getItem(SolConstantsUtil.CAIP_CHAIN_ID) : ''
+    ) as CaipNetwork
+    this.syncRequestedNetworks(chains, this.options?.chainImages)
+
     if (chain) {
       SolStoreUtil.setCurrentChain(chain)
       SolStoreUtil.setCaipChainId(`solana:${chain.chainId}`)
     }
-    this.syncNetwork(chainImages)
+    this.syncNetwork(this.options?.chainImages)
 
     SolStoreUtil.subscribeKey('address', () => {
       this.syncAccount()
     })
 
     SolStoreUtil.subscribeKey('caipChainId', () => {
-      this.syncNetwork(chainImages)
+      this.syncNetwork(this.instanceOptions?.chainImages)
     })
 
     AssetController.subscribeNetworkImages(() => {
-      this.syncNetwork(chainImages)
+      this.syncNetwork(this.instanceOptions?.chainImages)
     })
 
     NetworkController.subscribeKey('caipNetwork', (newCaipNetwork: CaipNetwork | undefined) => {
       const newChain = chains.find(_chain => _chain.chainId === newCaipNetwork?.id.split(':')[1])
 
       if (!newChain) {
-        throw new Error('The selected chain is not a valid Solana chain')
+        return
```

Comment:
Shouldn't we return an error in this situation? Make sure we are aligned with the version we have in `main`

Reply:
Correcting from my phone so hard to cross check but we ahould make sure throwing doesnt crash appkit

Reply:
This is probably going to behave weirdly, like if after trying to connect nothing happens. I can rework this later as I'm going to go over everything Solana related through the end of the week.